### DISCUSSION TITLE
Alpine Expression Error: X is not defined, when using wire:navigate

### DISCUSSION THREAD

**Discussion by EdgarsJoja** [livewire/livewire#8281] [7. Bugs (archived)] [answered]
### Livewire version

v3.4.10

### Laravel version

v11.0.8

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Have page (page 2) that uses `Alpine.data` for components as shown below.
e.g.
```
<div x-data="alpineDataId">
    {{-- some html --}}

    <script>
        (function () {
            const init = () => {
                Alpine.data('alpineDataId', () => {
                    //
                });
            };

            document.addEventListener('alpine:init', init);
            document.addEventListener('livewire:navigated', init);
        })();
    </script>
</div>
```

2. Do page reload on (page 1);
3. Go to page 2 (that contains component as shown above), using `wire:navigate`;
4. Observe error: "**Alpine Expression Error: alpineDataId is not defined**";

See Wirebox for specific example, this is mostly pseudo-code.

I'm not sure if this is the best approach how Alpine components are initialized (script, with self executing function, that adds event listeners for `alpine:init` and `livewire:navigated`). If there are some better approaches, that solve this issue, those are also welcome :).

Posting under bug section, due to issue specifically appearing in `v3.4.10`

### Repository or Wirebox URL

https://wirebox.app/b/4npn3

### Additional information

1. This issue seems to have appeared in `v3.4.10`. I downgraded my app to `v3.4.9` and the issue disappeared.
2. This seems to be related specifically to `wire:navigate`. Regular redirects have no issue as far as I can tell.

This issue seems to be somewhat similar https://github.com/livewire/livewire/discussions/6877, however, as I mentioned I had no problems with `v3.4.9`...

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment, marked as answer]
just wrap it inside `@script`

```blade
<div x-data="dataId"></div>

@script
    <script>
        Alpine.data('dataId', () => ({
            init() {
                console.log('Log from "Alpine.data"');
            }
        }));
    </script>
@endscript
```

**Reply by EdgarsJoja** [reply]
This seems to work indeed, thank you @danie-ramdhani . Somehow I had missed JS section in Livewire docs completely. I suppose, I tried the native plain html way initially, it somewhat worked, so never had to dig deeper...
The benefit of this approach also is that no longer self executing function with `alpine:init` and `livewire:navigated` is needed, as per docs:
> Because scripts inside @script are handled by Livewire, they are executed at the perfect time after the page has loaded, but before the Livewire component has rendered. This means you no longer need to wrap your scripts in document.addEventListener('...') to load them properly.
